pinot-segment-local/src/main/java/org/apache/pinot/segment/local/io/readerwriter/RealtimeIndexOffHeapMemoryManager.java

```diff
@@ -51,8 +51,8 @@ public abstract class RealtimeIndexOffHeapMemoryManager implements PinotDataBuff
   protected RealtimeIndexOffHeapMemoryManager(ServerMetrics serverMetrics, String segmentName) {
     _serverMetrics = serverMetrics;
     _segmentName = segmentName;
-    if (SegmentName.isLowLevelConsumerSegmentName(segmentName)) {
-      LLCSegmentName llcSegmentName = new LLCSegmentName(segmentName);
+    LLCSegmentName llcSegmentName = LLCSegmentName.getLLCSegmentName(segmentName);
+    if (llcSegmentName != null) {
       _tableName = llcSegmentName.getTableName();
```

Comment:
```suggestion
    if (LLCSegmentName.isLowLevelConsumerSegmentName(segmentName)) {
      _tableName = LLCSegmentName.of(segmentName).getTableName();
```

Reply:
This is intentional to avoid splitting the segment name twice